src/com/projectkorra/projectkorra/PKListener.java

```diff
@@ -1590,18 +1577,9 @@ public void onPlayerSwapItems(final PlayerSwapHandItemsEvent event) {
 		}
 	}
 
-	@EventHandler(priority = EventPriority.NORMAL)
-	public void onPlayerSwing(final PlayerInteractEvent event) {
+	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+	public void onPlayerSwing(final PlayerAnimationEvent event) {

```

Comment:
You should not be changing source code so that a single outdated addon can work.  Did you test these changes to see how they effect the stock abilities?

Reply:
It was already previously this beforehand in PK and was changed. 
For some reason, PlayerInteractEvent always returns false for left clicking air. He tried debugging loads to get a version that would work (lots of different other checks in the event for left clicking air and blocks) and changing it to PlayerAnimationEvent was the only thing that resolved this issue.

It isn't just to fix one addon but rather because this event shouldn't always be returning false when it isn't, which creates can more issues down the line,  if it doesn't already..

Reply:
If you look at the javadocs for API on PlayerInteractEvent, you'll see it states that it fires as cancelled if the vanilla behavior is to do nothing. 

Besides, PlayerAnimationEvent takes away our entire capability of checking between left and right click, throwing an item or not.

With these changes, it is completely possible to put a snowball in your offhand and by right clicking activate a FireBlast. This is not desirable behavior and it was changed in order to address this.

Reply:
Checking for left and right click should only need to be done for combos, not most moves. As it stands now, you cannot cancel bending even when the event is cancelled and that is heavily flawed. Things like snowballs & dropping items causing animations when thrown can be resolved other ways rather than changing everything to use the interact event

Reply:
^ what Strange said.

Even with it as PlayerInteractEvent, those issues you just raised still exist (I've been told there's a fix in WIP but still) 
We can find a workout for that. What we can't find a workaround is it always being cancelled.

Reply:
What I think @Varhagna trying to say is that this proposed change introduces issues into the codebase that this pull request does not address. Therefore, it cannot be accepted as is. Just because you want the immediate payoff of this change does not mean we are going to do all of the heavy lifting to get it implemented. We are more than happy to start a dialog on how we, together, might implement this change, otherwise, it will most likely end up on our internal todo list for evaluation and discussion involving the rework.